dotCMS/src/main/java/com/dotcms/rest/api/v1/workflow/SystemActionApiFireCommandFactory.java

```diff
@@ -243,4 +281,347 @@ private Contentlet firePublishWithSave(final Contentlet contentlet,
             return checkinContentlet;
         }
     }
+
+    //////////////////////////
+    /**
+     * Implements a {@link SystemActionApiFireCommand} that does an unpublish to cover the {@link com.dotmarketing.portlets.workflows.business.WorkflowAPI.SystemAction#UNPUBLISH} system action
+     */
+    private class UnpublishSystemActionApiFireCommandImpl implements SystemActionApiFireCommand {
+
+        @WrapInTransaction
+        @Override
+        public Contentlet fire(final Contentlet contentlet, final boolean needSave, final ContentletDependencies dependencies)
+                throws DotDataException, DotSecurityException {
+
+            Logger.info(this, "The contentlet : " + contentlet.getTitle()
+                    + ", was fired by default action: " + dependencies.getWorkflowActionId() +
+                    ", however this action has not any unpublish content actionlet, so the unpublish api call is being triggered as part of the request");
+
+            final String actionId       = UtilMethods.isSet(dependencies.getWorkflowActionId())?
+                    dependencies.getWorkflowActionId():contentlet.getActionId();
+            final User   user           = dependencies.getModUser();
+
+            if(UtilMethods.isSet(actionId)) {
+                contentlet.setActionId(actionId);
+            }
+
+            if(UtilMethods.isSet(dependencies.getWorkflowActionComments())){
+                contentlet.setStringProperty(Contentlet.WORKFLOW_COMMENTS_KEY, dependencies.getWorkflowActionComments());
+            }
+
+            if(UtilMethods.isSet(dependencies.getWorkflowAssignKey())){
+                contentlet.setStringProperty(Contentlet.WORKFLOW_ASSIGN_KEY, dependencies.getWorkflowAssignKey());
+            }
+
+            if (needSave) {
+
+                return fireUnPublishWithSave(contentlet, dependencies, actionId, user);
+            }
+
+            Logger.info(this, "The contentlet : " + contentlet.getTitle()
+                    + ", on the action id: " + actionId +
+                    ", will do an unpublish");
+
+            contentletAPI.unpublish(contentlet, user, dependencies.isRespectAnonymousPermissions());
+
+            return contentlet;
+        }
+
+        private Contentlet fireUnPublishWithSave(final Contentlet contentlet,
+                                               final ContentletDependencies dependencies,
+                                               final String actionId, final User user) throws DotDataException, DotSecurityException {
+
+            final String disableWorkflow         = Contentlet.DISABLE_WORKFLOW;
+            final boolean hasUnpublishActionlet  = UtilMethods.isSet(actionId)?
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Ternary operators that can be simplified with || or &&](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4324168)